### ISSUE TITLE
POSSIBLE BUG: addnodes.literal_emphasis and addnodes.literal_strong don't inherit from nodes.literal

### ISSUE THREAD

**Issue by bskinn** [sphinx-doc/sphinx#3787]
In the course of investigating #3781, I noticed that [`addnodes.literal_emphasis` and `addnodes.literal_strong`](https://github.com/sphinx-doc/sphinx/blob/8f9dd6499207be2f3291a0982ed02e418730ef5e/sphinx/addnodes.py#L269-L278) do not subclass from `docutils.nodes.literal`. Thus, AFAICT they do not render as literal text.  Is this intended behavior?

**Comment by jfbu** [comment]
The SmartyPantsHTMLTranslator class from html writer, removed at #3666, handled indeed especially these nodes, by setting ``self.no_smarty += 1``. Unfortunately the switch to Docutils's processing of smart quotes overlooked that. (Docutils looks only for

```
                                           (nodes.literal,
                                            nodes.math,
                                            nodes.image,
                                            nodes.raw,
                                            nodes.problematic))]
```
 
as nodes type skipping the SmartQuotes.

It is not clear to me if ``literal_emphasis`` should inherit from ``literal``, but definitely there is a bug here introduced at 1.6.1. Thanks for the finding.

**Comment by mitya57** [comment]
+1 for making these inherit from `nodes.literal`. Then we would be able to remove two lines in `SphinxSmartQuotes.get_tokens`.

@jfbu maybe you will add this to your PR #3781? (simply change i.e. `literal_emphasis(nodes.emphasis)` to `literal_emphasis(nodes.emphasis, nodes.literal)`).

**Comment by jfbu** [comment]
@mitya57 unfortunately I don't have a clear view of the semantics of ``literal_emphasis``. Making it inherit from `literal` now makes one wonder why one did not earlier... indeed, earlier situation caused all writers to handle especially, for example html added the ``no_smarty`` flag. 

I need some advice by further knowledgeable Sphinx people ;-) (also my lunch is burning on my oven at taht time... will be back later)

**Comment by tk0miya** [comment]
Fixed at #3816.
